File: C/Easy/2264_Largest_3_Same_Digit_Number_in_String/solution.c

char* largestGoodInteger(char* num) {
    int maxCount = -1;

    for (int i = 1; num[i+1] != '\0'; i++ ) {
        if (num[i-1] == num[i] && num[i] == num[i+1]) {
            int v = num[i] - '0';
            if (v > maxCount) {
                maxCount = v;
            }
        }
    }

    if (maxCount == -1) {
        return "";
    }

    static char charDigit[4];
    charDigit[0] = maxCount + '0';
    charDigit[1] = maxCount + '0';
    charDigit[2] = maxCount + '0';
    charDigit[3] = '\0';
    return charDigit;
}

File: Golang/Easy/2264_Largest_3_Same_Digit_Number_in_String/solution.go
package largest3samedigitnumberinstring

import "strconv"

func largestGoodInteger(num string) string {
	maxCount := -1
	for i := 1; i < len(num)-1; i++ {
		if num[i-1] == num[i] && num[i] == num[i+1] {
			n, _ := strconv.Atoi(string(num[i]))
			maxCount = max(maxCount, n)
		}
	}

	if maxCount == -1 {
		return ""
	}
	numString := strconv.Itoa(maxCount)
	return numString + numString + numString
}
